website/api/controllers/webhooks/receive-github-projects-v2-item.js

```diff
@@ -0,0 +1,984 @@
+ *
+ * 2. "Awaiting QA" status:
+ *    - Triggers when: Status changes TO "awaiting qa"
+ *    - From states:
+ *      - "in progress" or "review" → Always creates new row
+ *      - Any other state → Creates row only if no QA ready row exists
+ *    - Saves to: github_metrics.issue_qa_ready
+ *    - Data: qa_ready time, assignee, issue details, time from in_progress to qa_ready
+ *    - Note: Requires an existing in_progress record to calculate time
+ *
+ * 3. "Release" status:
+ *    - Triggers when: Status changes TO "release"
+ *    - From states:
+ *      - "awaiting qa" → Always creates new row
+ *      - Any other state → Creates row only if issue has been in QA (exists in issue_qa_ready)
+ *    - Saves to: github_metrics.issue_release_ready
+ *    - Data: release_ready time, assignee, issue details, time from in_progress to release_ready
+ *    - Note: Requires an existing in_progress record to calculate time
+ *
+ * Time calculations:
+ * - All time calculations can optionally exclude weekends (controlled by EXCLUDE_WEEKENDS flag)
+ * - Weekend exclusion adjusts start/end times and subtracts weekend days from duration
+ * - Times are calculated from the webhook's updated_at timestamp for accuracy
+ *
+ * Issue type classification:
+ * - Based on GitHub issue labels:
+ *   - "bug" → type: "bug"
+ *   - "story" → type: "story"
+ *   - "~sub-task" → type: "sub-task"
+ *   - Otherwise → type: "other"
+ */
+
+// Dependencies
+const {BigQuery} = require('@google-cloud/bigquery');
+const crypto = require('crypto');
+
+// Project constants
+const PROJECTS = {
+  ORCHESTRATION: 71,
+  MDM: 58,
+  SOFTWARE: 70
+};
+
+// Flag for excluding weekends in time calculations. Only used during dev.
+// Set to true to exclude weekends, false to include them
+const EXCLUDE_WEEKENDS = true;
+
+// BigQuery client (initialized on first use)
+let bigqueryClient = null;
+
+module.exports = {
+
+  friendlyName: 'Receive GitHub Projects v2 item',
+
+  description: 'Receive webhook requests from GitHub for Projects v2 item events.',
+
+  extendedDescription: 'This webhook endpoint receives JSON data from GitHub when project items are created, updated, or deleted.',
+
+  inputs: {
+    // GitHub sends the entire payload as JSON in the request body
+    // We'll capture all the data dynamically
+  },
+
+  exits: {
+    success: {
+      description: 'Webhook processed successfully'
+    },
+    unauthorized: {
+      description: 'Invalid or missing webhook signature',
+      responseType: 'unauthorized'
+    }
+  },
+
+  fn: async function () {
+
+    // Check if webhook secret is configured
+    if (!sails.config.custom.githubProjectsV2ItemWebhookSecret) {
+      throw new Error('No GitHub Projects v2 item webhook secret configured! (Please set `sails.config.custom.githubProjectsV2ItemWebhookSecret`.)');
+    }
+
+    // Verify webhook signature
+    verifyGitHubWebhookSignature(this.req, sails.config.custom.githubProjectsV2ItemWebhookSecret);
```

Comment:
If we use the other webhook action (receive-from-github), then webhook authentication is already built in.